### PR TITLE
community[patch]: restore compatibility with SQLAlchemy 1.x

### LINKED ISSUE #19681: pgvector.py has created a dependency on  sqlalchemy v2, breaking apps using sqlalchemy v1
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

from langchain.retrievers.multi_query import MultiQueryRetriever

### Error Message and Stack Trace (if applicable)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/miniconda3/envs/nettalk/lib/python3.12/site-packages/langchain/retrievers/__init__.py", line 33, in <module>
    from langchain.retrievers.self_query.base import SelfQueryRetriever
  File "/home/ubuntu/miniconda3/envs/nettalk/lib/python3.12/site-packages/langchain/retrievers/self_query/base.py", line 5, in <module>
    from langchain_community.vectorstores import (
  File "/home/ubuntu/miniconda3/envs/nettalk/lib/python3.12/site-packages/langchain_community/vectorstores/__init__.py", line 115, in __getattr__
    module = importlib.import_module(_module_lookup[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/miniconda3/envs/nettalk/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/miniconda3/envs/nettalk/lib/python3.12/site-packages/langchain_community/vectorstores/pgvector.py", line 23, in <module>
    from sqlalchemy import SQLColumnExpression, delete, func
ImportError: cannot import name 'SQLColumnExpression' from 'sqlalchemy' (/home/ubuntu/miniconda3/envs/nettalk/lib/python3.12/site-packages/sqlalchemy/__init__.py)

### Description

My app uses sqlalchemy v1 due to my db dialect not supporting v2.  The above change breaks MultiQueryRetriever in 0.1.13.

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #17~22.04.1-Ubuntu SMP Fri Nov 17 21:07:13 UTC 2023
> Python Version:  3.12.2 | packaged by conda-forge | (main, Feb 16 2024, 20:50:58) [GCC 12.3.0]

Package Information
-------------------
> langchain_core: 0.1.33
> langchain: 0.1.13
> langchain_community: 0.0.29
> langsmith: 0.1.31
> langchain_experimental: 0.0.55
> langchain_openai: 0.1.1
> langchain_text_splitters: 0.0.1
> langgraph: 0.0.25

Packages not installed (Not Necessarily a Problem)
--------------------------------------------------
The following packages were not found:

> langserve

### PR DESCRIPTION
**Pull request by krassowski**
- **Description:** Restores compatibility with SQLAlchemy 1.4.x that was broken since #18992 and adds a test run for this version on CI (only for Python 3.11)
- **Issue:** fixes #19681
- **Dependencies:** None
- **Twitter handle:** `@krassowski_m`

### CODE REVIEW

File: .github/workflows/_test.yml [hunk 1]
```diff
@@ -28,7 +28,12 @@ jobs:
           - "3.9"
           - "3.10"
           - "3.11"
-    name: "make test #${{ matrix.python-version }}"
+        sqlalchemy-version:
```
Comment (eyurtsev) [inline .github/workflows/_test.yml:31]: @efriis let me know what you think.

We've had multiple breakages with users using sqlalchemy 1 or 2 for dev work, so adding some support in CI would be useful

File: .github/workflows/_test.yml [hunk 2]
```diff
@@ -28,7 +28,12 @@ jobs:
...
+          - "latest"
+        include:
+          - sqlalchemy-version: "1.4"
```
Comment (eyurtsev) [inline .github/workflows/_test.yml:34]: Potentially fragile since it assumes that the lock file has sqlalchemy 2?

Comment (krassowski) [reply .github/workflows/_test.yml:34]: Maybe rewording the default for `sqlalchemy-version` from "latest" to "default" or "from-lock-file" would make it more acceptable by making it explicit?

If not, I can specify both ">2" and "~=1.4" and adjust the logic for not polluting the name needlessly with matching against ">2".

Do let me know what change if any you prefer :slightly_smiling_face:

Comment (eyurtsev) [reply .github/workflows/_test.yml:34]: @krassowski assigning to @efriis since he's been setting up all the devs ops

Comment (krassowski) [reply .github/workflows/_test.yml:35]: Just checking back here one week later :)

**Comment by anilbmx** [comment]
This fix will help a lot !!. I'm trying to use latest version of langchain in airflow but they haven't upgraded the dependency which is still using 1.x version.
Hope they approve this :-)

**Review by efriis — APPROVED** [review]
Going to remove the CI change for now and will discuss when I get back from vacation! I think we should add it to minimum version testing on releases, and potentially add the unit test version of it to pr CI (and also for langchain-core, and any other libraries we add to that min version checking file)
